st2actions/tests/unit/policies/test_concurrency_by_attr.py

```diff
@@ -99,25 +99,24 @@ def test_over_threshold_delay_executions(self):
 
         scheduled = [item for item in LiveAction.get_all() if item.status in SCHEDULED_STATES]
         self.assertEqual(len(scheduled), policy_db.parameters['threshold'])
+        self.assertEqual(len(scheduled), runner.MockActionRunner.run.call_count)
 
         # Execution is expected to be delayed since concurrency threshold is reached.
         liveaction = LiveActionDB(action='wolfpack.action-1', parameters={'actionstr': 'fu'})
         liveaction, _ = action_service.request(liveaction)
 
-        # Assert the delayed state is being published.
-        calls = [call(liveaction, action_constants.LIVEACTION_STATUS_DELAYED)]
```

Comment:
Perhaps some more "explicit" tests (explicitly asserting that call count is zero for non-cancel publish method call) would make it more obvious and harder in the future to make the same mistake.

A comment in the tests would also be nice.